source/app.d

```diff
@@ -248,6 +297,121 @@ void updateGithubComment(string action, IssueRef[] refs, Issue[] descs, string c
+    string commitsURL;
+    string commentsURL;
+    string pullRequestURL;
+}
+
+alias LabelsAndCommits = Tuple!(Json[], "labels", Json[], "commits");
+alias LabelInfo = Tuple!(bool, "hasAutoMerge", bool, "hasAutoMergeSquash");
+
+auto analyseLabels(Json[] labels)
+{
+    auto labelNames = labels.map!(l => l["name"].get!string).array;
+    bool hasAutoMerge = labelNames.any!(l => l == "auto-merge");
+    bool hasAutoMergeSquash = labelNames.any!(l => l == "auto-merge-squash");
+    return LabelInfo(hasAutoMerge, hasAutoMergeSquash);
+}
+
+auto handleGithubLabel(in ref PullRequestParams pr)
+{
+    auto url = "%s/repos/%s/issues/%d/labels".format(githubAPIURL, pr.repoSlug, pr.pullRequestNumber);
+    auto res = ghGetRequest(url);
+
+    auto labels = res.readJson[];
+    auto labelInfo = analyseLabels(labels);
+    Json[] commits;
+
+    if (labelInfo.hasAutoMerge || labelInfo.hasAutoMergeSquash)
+        commits = pr.tryMerge(labelInfo);
+
+    return LabelsAndCommits(labels, commits);
+}
+
+Json[] tryMerge(in ref PullRequestParams pr, LabelInfo labelInfo)
+{
+    auto commits = ghGetRequest(pr.commitsURL).readJson[];
+
+    if (!pr.isOpen)
+    {
+        logWarn("Can't auto-merge PR %s/%d - it is already closed", pr.repoSlug, pr.pullRequestNumber);
+        return commits;
+    }
+
+    if (commits.length == 0)
+    {
+        logWarn("Can't auto-merge PR %s/%d has no commits attached", pr.repoSlug, pr.pullRequestNumber);
+        return commits;
+    }
+
+    string lastCommitSha = commits[$ - 1]["sha"].get!string;
+    string[string] reqInput = ["sha": lastCommitSha];
+
+    // lazy implementation -> try merge
+    if (labelInfo.hasAutoMerge)
+        reqInput["merge_method"] = "merge";
+
+    if (labelInfo.hasAutoMergeSquash)
+        reqInput["merge_method"] = "squash";
+
+    auto prUrl = "%s/repos/%s/pulls/%d/merge".format(githubAPIURL, pr.repoSlug, pr.pullRequestNumber);
+    ghSendRequest((scope req){
+        req.method = HTTPMethod.PUT;
+        // custom media type is required during preview period:
+        // https://developer.github.com/changes/2016-09-26-pull-request-merge-api-update/
+        req.headers["Accept"] = "application/vnd.github.polaris-preview+json";
+        req.writeJsonBody(reqInput);
+    }, prUrl);
+
+    return commits;
+}
+
+void checkAndRemoveMergeLabels(Json[] labels, in ref PullRequestParams pr)
+{
+    auto labelInfo = analyseLabels(labels);
+    if (labelInfo.hasAutoMerge || labelInfo.hasAutoMergeSquash)
+    {
+        auto labelUrl = "%s/repos/%s/issues/%d/labels/".format(githubAPIURL, pr.repoSlug, pr.pullRequestNumber);
+        if (labelInfo.hasAutoMerge)
+            labelUrl ~= "auto-merge";
+        if (labelInfo.hasAutoMergeSquash)
+            labelUrl ~= "auto-merge-squash";
+        ghSendRequest(HTTPMethod.DELETE, labelUrl);
+    }
+}
+
+void searchForAutoMergePrs(string repoSlug)
+{
+    auto issues = ghGetRequest("%s/repos/%s/issues?state=open&labels=auto-merge,auto-merge-squash"
```

Comment:
Unfortunately this is a logical AND between the labels, see https://api.github.com/repos/dlang/phobos/issues?state=open&labels=blocked,dependency. Searched the net, there doesn't seem to be an OR, so we'd need 2 queries (and maybe an intersection :/).